src/py/flwr/server/grpc_server/flower_service_servicer.py

```diff
@@ -102,8 +132,29 @@ def Join(  # pylint: disable=invalid-name
                     # Get ins_wrapper from bridge and yield server_message
                     ins_wrapper: InsWrapper = next(ins_wrapper_iterator)
                     yield ins_wrapper.server_message
+
                     # Wait for client message
-                    client_message = next(client_message_iterator)
+                    client_message = next_with_timeout(
+                        iterator=client_message_iterator,
+                        timeout=ins_wrapper.timeout,
+                    )
+                    if client_message is None:
+                        # Important: calling `context.abort` in gRPC always
+                        # raises an exception so that all code after the call to
+                        # `context.abort` will not run. If subsequent code should
+                        # be executed, the `rpc_termination_callback` can be used
+                        # (as shown in the `register_client` function).
```

Comment:
```suggestion
                    if client_message is "None":
                        # Important: calling `context.abort` in gRPC always
                        # raises an exception so that all code after the call to
                        # `context.abort` will not run. If subsequent code should
                        # be executed, the `rpc_termination_callback` can be used
                        # (as shown in the `register_client` function).
```